### PR TITLE
Bump tutor requirements to allow v17

### PR DESCRIPTION
**Pull request by gabor-boros**
## Description

This PR bumps Tutor to v17 to support Quince. Also, it updates the changelog and the compatibility matrix.

### CODE REVIEW

**Comment by gabor-boros** [comment]
FYI @fghaas

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## Unreleased
+
+* [Enhancement] Support Tutor 17 and Open edX Quince
```
Comment (fghaas) [inline CHANGELOG.md:3]: Add full stop (`.`) at the end please. :)

File: README.md [hunk 2]
```diff
@@ -36,7 +37,7 @@ appropriate one:
 Installation
 ------------
 
-    pip install git+https://github.com/hastexo/tutor-contrib-s3@v1.2.0
+    pip install git+https://github.com/hastexo/tutor-contrib-s3@v1.3.0
```
Comment (fghaas) [inline README.md:40]: Exclude this change, please. This goes into a separate commit created by `bump2version` when we cut the release.

**Review by fghaas — COMMENTED** [review]
Looks good to me; I just have two nits that I'd ask you to fix.

Do I assume correctly that you have already tested this, on Quince?

**Comment by fghaas** [comment]
Let me check on the build failures from `flake8`.

**Comment by fghaas** [comment]
@gabor-boros #67 is now merged; when you update your topic branch please be sure to rebase it on current `main`.

**Comment by gabor-boros** [comment]
@fghaas Thanks for the review. I'm going to address the changes soon. The quince compatibility was not yet tested however the new version of tutor should not break this plugin as far as I know. Though I'll do a test round.

**Comment by gabor-boros** [comment]
Hey @fghaas. We are currently blocked on https://github.com/overhangio/tutor/pull/958

**Comment by gabor-boros** [comment]
@fghaas The changes are tested, the plugin is working and the upstream issue is resolved as well. This is ready to be reviewed and merged.

**Comment by fghaas** [comment]
@gabor-boros Hold on, I don't quite follow. If this plugin broke on Quince as long as https://github.com/overhangio/tutor/pull/958 wasn't merged, then that means it would break on 17.0.0. So, don't we have to wait until a 17.0.1 release to get the dependencies right?

**Comment by gabor-boros** [comment]
Sorry for the confusion. Let me clarify:

The plugin was never broken, but we were not able to test it as the nightly branch of Tutor was broken. This plugin perfectly works on Quince, I just tested it today.

**Comment by fghaas** [comment]
Ah, gotcha! Sounds good to me then.

Can I bother you for one more amend and force-push, so we get a meaningful commit message body? Something like

```
chore: Bump supported Tutor versions to work on Tutor 17 for Quince
    
Update the install_requires list to include support for Tutor 17 (and
Open edX Quince), and update the README in anticipation of a 1.3.0
release.
    
Signed-off-by: Gabor Boros <gabor.brs@gmail.com>
```

I would much appreciate that. Thanks!

**Comment by gabor-boros** [comment]
@fghaas how about now?

**Review by fghaas — APPROVED** [review]
Perfect. Thanks a lot!

**Comment by fghaas** [comment]
@gabor-boros Could you do me a huge favour and have a brief look into #70? I am a bit surprised that you didn't run into that during your tests, and would like to find out why not.

**Comment by gabor-boros** [comment]
@fghaas I'm surprised as well, though there were some dependency caching issues on our end around the time of this PR, so probably the dependencies were just simply not using the new package version.

Though this change is 2 months old, so I cannot recall exactly.

**Pull request merged**